packages/x-components/src/components/base-variable-column-grid.vue

```diff
@@ -26,69 +26,77 @@
-    /**
-     * Animation component that will be used to animate the grid.
-     *
-     * @public
-     */
-    @Prop({ default: 'ul' })
-    protected animation!: Vue | string;
-
-    /**
-     * The list of items to be rendered.
-     *
-     * @remarks The items must have an id property.
-     *
-     * @public
-     */
-    @Prop()
-    protected items?: ListItem[];
-
-    /**
-     * The columns to render by default in the grid. This property is used when the user has not
-     * selected any value in the column picker.
-     *
-     * @internal
-     */
-    @Prop({ default: 0 })
-    protected columns!: number;
-
-    /**
-     * The number of columns provided by a user interaction.
-     *
-     * @internal
-     */
-    protected providedColumns: number | null = null;
-
-    /**
-     * The number of columns to render in the grid.
-     *
-     * @returns The number of columns.
-     *
-     * @internal
-     */
-    protected get columnsToRender(): number {
-      return this.providedColumns === null ? this.columns : this.providedColumns;
-    }
+    },
+    props: {
+      /**
+       * Animation component that will be used to animate the grid.
+       *
+       * @public
+       */
+      animation: {
+        type: AnimationProp,
+        default: 'ul'
+      },
+      /**
+       * The list of items to be rendered.
+       *
+       * @remarks The items must have an id property.
+       *
+       * @public
+       */
+      items: Array as PropType<ListItem[]>,
+      /**
+       * The columns to render by default in the grid. This property is used when the user has not
+       * selected any value in the column picker.
+       *
+       * @internal
+       */
+      columns: {
+        type: Number,
+        default: 0
+      }
+    },
+    setup(props, { slots }) {
+      const bus = useXBus();
+      /**
+       * The number of columns provided by a user interaction.
+       *
+       * @internal
+       */
+      const providedColumns = ref<number | null>(null);
 
-    /**
-     * Handler to update the number of columns when the user selects a new value.
-     *
-     * @param newColumns - The new columns value.
-     *
-     * @internal
-     */
-    @XOn(['ColumnsNumberProvided'])
-    setColumns(newColumns: number): void {
-      this.providedColumns = newColumns;
+      /**
+       * The number of columns to render in the grid.
+       *
+       * @returns The number of columns.
+       *
+       * @internal
+       */
+      const columnsToRender = computed(() =>
+        providedColumns.value === null ? props.columns : providedColumns.value
+      );
+
+      /**
+       * Handler to update the number of columns when the user selects a new value.
+       *
+       * @param newColumns - The new columns value.
+       *
+       * @internal
+       */
+      bus
+        .on('ColumnsNumberProvided', false)
+        .subscribe((newColumns: number) => (providedColumns.value = newColumns));
```

Comment:
`newColumns` type is already inferred. You can remove it
![Screenshot 2024-05-27 at 6 54 04 PM](https://github.com/empathyco/x/assets/10746604/44e96197-cf03-4fdc-adc0-d7250275f195)